README.md

```diff
@@ -164,9 +163,117 @@ modelSelector | JSON | Use the Cloudant Query selector syntax to associate model
-  
-Considering the nature of Cloudant, we plan to give user the flexibility to define `_rev` property in model and provide `_rev` by themselves instead of the connector. In such way user can get the revision and control conflict before using that revision to modify the document.
+In a document, property `_rev` is the latest doc revision and must be provided when modifying the doc.
+
+Our connector allows the user to retrieve back the `_rev` property upon all CRUD operations, however does not add it to the model definition.
+
+If you would like to have a `_rev` property on your model, as an end user, the onus is on you to add the property in the model definition.
+
+#### Example CRUD operations with `_rev`
+
+`model.json`
+``` json
+{
+  ... 
+  "properties": {
+    "_rev": {
+      "type": "string"
+    },
+    "name": {
+      "type": "string"
+    }
+  },
+  ...
+}
+```
+
+- Create
+
+```javascript
+  Model.create([{
+    name: 'Foo',
+  }, {
+    name: 'Bar',
+  }], function(err, result) {
+    if (err) throw err;
+    console.log('Created instance: ' + JSON.stringify(result));
+  });
+```
+
+**Note:** Cloudant does not allow customized `_rev` value, hence creating an instance with a `_rev` value will give an error. The onus is on the user if they fail to comply to this rule.
+
+Let's say we have an instance in the database:
+```json
+{
+   "id":"2",
+   "_rev":"2-abcedf",
+   "name":"Bar"
+}
+```
+
+- Find
+
+  - find
+
+  ```javascript
+    Model.findById('2', function(err, result) {
+      if (err) throw err;
+      console.log('Found instance with id: ' + JSON.stringify(result));
+    });
+  ```
+
+  - findById
+
+  ```javascript
+    Model.find(function(err, result) {
+      if (err) throw err;
+      console.log('Found all instances: ' + JSON.stringify(result));
+    });
+  ```
+
+- Replace
+
+  - replaceOrCreate
+
+  ```javascript
+    Model.replaceOrCreate({
+      id:'2',
+      _rev:'2-abcedf',
+      name:'Bar2'
+    }, function(err, result) {
+      if (err) throw err;
+      console.log('Replace an existing instance: ' + JSON.stringify(result));
+    });
+  ```
+
+  - replaceById
+
+  ```javascript
+    Model.replaceById('2', {
+      _rev:'2-abcedf',
+      name:'Bar3'
+    }, function(err, result) {
+      if (err) throw err;
+      console.log('Replace an existing instance with id: ' + JSON.stringify(result));
+    });
+  ```
+
+- Update
+
+  - updateOrCreate
+
+  ```javascript
+    Model.updateOrCreate({
+      id:'2',
+      _rev:'2-abcedf',
+      name:'Bar4'
+    }, function(err, result) {
+      if (err) throw err;
+      console.log('Update an existing instance: ' + JSON.stringify(result));
+    });
+  ```
+
+**Note:** When calling either `replaceOrCreate` or `updateOrCreate` function, **_do not specify the `_rev` value**_, unless the user is confident it should do a `replace`/`update` instead of a `create`. This is because Cloudant does not allow custom `_rev` value on a `create` operation.
```

Comment:
@kjdelisle Fixed this. Now the user can call `replaceOrCreate`, `updateOrCreate`, `replaceById`, `update` and send out a _rev value. We strip it out before making the final _insert call.